ext/js/display/media-drawing-worker.js

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2024  Yomitan Authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import {API} from '../comm/api.js';
+import {createApiMap, invokeApiMapHandler} from '../core/api-map.js';
+import {log} from '../core/log.js';
+import {WebExtension} from '../extension/web-extension.js';
+
+export class MediaDrawingWorker {
+    constructor() {
+        /** @type {number} */
+        this._generation = 0;
```

Comment:
`generation` is a fairly common term used when you have some sort of garbage collection going on -- and we do here, as we clear all the old canvases from memory when there is a new lookup occurring. This is required because the backend may still be processing old requests and sending some instructions to draw things on indexes for certain canvases from the previous lookup which could cause bugs if we didn't have this tracking of generations.